client-runtime/src/main/java/com/microsoft/rest/v2/SwaggerMethodProxyDetails.java

```diff
@@ -233,7 +233,7 @@ public void setReturnType(Class<?> returnType) {
      * object that is returned from the method.
      * @return The synchronous return type for the method that this object describes.
      */
-    public Class<?> getReturnType() {
+    public Class<?> returnType() {
```

Comment:
Type is an interface that Class implements, so it doesn't necessarily give us any more information to use Type instead of Class (as far as I understand). Also, I used a debugger to step through our unit tests and I wasn't able to get more generic type information by casting it to other Type classes (ParameterizedType, Type, etc.). Maybe we should talk some more about this though if you know more about it.